### PR TITLE
update overwatch version

### PR DESCRIPTION
**Pull request by al8n**

### CODE REVIEW

File: nodes/nomos-node-api/src/lib.rs [hunk 1]
```diff
@@ -62,9 +61,14 @@ where
...
-        let endpoint = B::new(self.settings.backend_settings).await?;
-        endpoint.serve(self.handle).await?;
+    async fn run(mut self) -> Result<(), ServiceError> {
+        let endpoint = B::new(self.settings.backend_settings)
+            .await
+            .map_err(|e| ServiceError::Service(Box::new(e)))?;
+        endpoint
+            .serve(self.handle)
+            .await
+            .map_err(|e| ServiceError::Service(Box::new(e)))?;
```
Comment (danielSanchezQ) [inline nodes/nomos-node-api/src/lib.rs:71]: One thing that we can do is to `impl From DynErr for ServiceError`, that way we would not need to `map_err` constantly I think. Wdyt @al8n ? (this needs to be done in the overwatch branch).

File: nomos-cli/src/da/disseminate.rs [hunk 2]
```diff
@@ -154,7 +151,8 @@ impl ServiceCore for DisseminateService {
             output,
         } = service_state.settings_reader.get_updated_settings();
 
-        let da_protocol: FullReplication<_> = da_protocol.try_into()?;
+        let da_protocol: Result<FullReplication<_>, _> = da_protocol.try_into();
+        let da_protocol = da_protocol.map_err(ServiceError::custom)?;
```
Comment (danielSanchezQ) [inline nomos-cli/src/da/disseminate.rs:155]: This can be chained.

File: nomos-services/http/examples/graphql.rs [hunk 3]
```diff
@@ -85,7 +85,7 @@ async fn handle_graphql_req(
     >,
     payload: Option<Bytes>,
     dummy: OutboundRelay<DummyGraphqlMsg>,
-) -> Result<String, overwatch_rs::DynError> {
+) -> Result<String, Box<dyn Error + Send + Sync + 'static>> {
```
Comment (danielSanchezQ) [inline nomos-services/http/examples/graphql.rs:88]: Why?

File: nomos-services/http/examples/graphql.rs [hunk 4]
```diff
@@ -179,7 +179,7 @@ pub struct Args {
     http: AxumBackendSettings,
 }
 
-fn main() -> Result<(), overwatch_rs::DynError> {
+fn main() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
```
Comment (danielSanchezQ) [inline nomos-services/http/examples/graphql.rs:182]: Did we removed `DynError` completely?

File: nodes/nomos-node-api/src/lib.rs (hunk 1 above)
Comment (al8n) [reply nodes/nomos-node-api/src/lib.rs]: `From DynErr` not works so good, rust compiler is not smart enough, we have to use 
```rust
.map_err(|e| Box::new(e) as DynErr)?
```

File: nomos-services/http/examples/graphql.rs (hunk 4 above)
Comment (al8n) [reply nomos-services/http/examples/graphql.rs]: no, I change it back to `DynError`

File: nodes/nomos-node-api/src/lib.rs (hunk 1 above)
Comment (danielSanchezQ) [reply nodes/nomos-node-api/src/lib.rs]: Uhm, let me take a shot at it.

File: nomos-services/http/src/http.rs [hunk 5]
```diff
@@ -188,25 +188,44 @@ pub async fn handle_graphql_req<M, F>(
...
+    #[derive(Debug)]
+    struct EmptyPayload;
+
+    impl core::fmt::Display for EmptyPayload {
+        fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+            write!(f, "empty payload")
+        }
+    }
+
+    impl Error for EmptyPayload {}
```
Comment (danielSanchezQ) [inline nomos-services/http/src/http.rs:207]: I do not think this impls should be inside the function itself. Any reason why?

File: nodes/nomos-node-api/src/lib.rs [hunk 6]
```diff
@@ -62,9 +62,14 @@ where
     }
 
     /// Service main loop
-    async fn run(mut self) -> Result<(), DynError> {
-        let endpoint = B::new(self.settings.backend_settings).await?;
-        endpoint.serve(self.handle).await?;
+    async fn run(mut self) -> Result<(), ServiceError> {
+        let endpoint = B::new(self.settings.backend_settings)
+            .await
+            .map_err(|e| Box::new(e) as DynError)?;
```
Comment (danielSanchezQ) [inline nodes/nomos-node-api/src/lib.rs:68]: Maybe we could make a method? so we do `map_err(ServiceError::from_whatever)`? Which can be generic?

File: nomos-services/http/src/http.rs (hunk 5 above)
Comment (al8n) [reply nomos-services/http/src/http.rs]: Yeah, at that moment, we do not have `Custom` for `ServiceError`. Now I add a `ServiceError::custom` fn to solve this.

File: nodes/nomos-node-api/src/lib.rs (hunk 6 above)
Comment (al8n) [reply nodes/nomos-node-api/src/lib.rs]: Yeah, I change it back to `ServiceError::service`.

**Pull request closed without merging**